File: Year_2/Semester_1/Metode_Avansate_De_Programare/Lab_3-6/MAP_ToySocialNetwork/MAP_ToySocialNetwork/src/Exception/RepositoryException/RepositoryException.java
package Exception.RepositoryException;

import Repository.Repository;

public class RepositoryException extends RuntimeException{
    /**
     * class dedicated to repository exceptions.
     */
    public RepositoryException()
    {

    }
    public RepositoryException(String message)
    {
        super(message);
    }
    public RepositoryException(String message,Throwable cause)
    {
        super(message, cause);
    }
    public RepositoryException(Throwable cause) {
        super(cause);
    }

    public RepositoryException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/C#_Project/Laboratory_2/FestivalTicketSales/Repository/Exceptions/RepositoryException.cs
namespace FestivalTicketSale.Repository.Exceptions;

public class RepositoryException : Exception
    {
        public RepositoryException() : base()
        {

        }

        /// <summary>
        /// Constructor with a message parameter
        /// </summary>
        /// <param name="message">Message to throw</param>
        public RepositoryException(string message) : base(message)
        {

        }
    }